common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -393,6 +394,35 @@ public void applicationRemoved(String appId, boolean cleanupLocalDirs) {
     }
   }
 
+  @Override
+  public void removeShuffleMerge(RemoveShuffleMerge msg) {
+    String appId = msg.appId;
+    int appAttemptId = msg.appAttemptId;
+    int shuffleId = msg.shuffleId;
+    AppShuffleInfo appShuffleInfo = validateAndGetAppShuffleInfo(appId);
+    if (appShuffleInfo.attemptId != appAttemptId) {
+      throw new IllegalArgumentException(
+          String.format("The attempt id %s in this RemoveShuffleMerge message does not match "
+                  + "with the current attempt id %s stored in shuffle service for application %s",
+              appAttemptId, appShuffleInfo.attemptId, appId));
+    }
+
+    appShuffleInfo.shuffles.compute(shuffleId, (shuffleIdKey, partitionsInfo) -> {
+      if (null != partitionsInfo) {
```

Comment:
The validation related to `shuffleMergeId`, from `finalizeShuffleMerge`, is applicable here depending on `shuffleMergeId`.
I have given the details [here](https://github.com/apache/spark/pull/37922#discussion_r990753031) - please refer to it.